### DISCUSSION TITLE
Share built-in USB-JTAG interface between Esptool and REPL

### DISCUSSION THREAD

**Discussion by Nico-GYW** [micropython/micropython#10643] [ESP32]
Hello everyone,

I recently installed the Micropython firmware on my esp32-S3 via the built-in USB-JTAG interface (D+/D- on GPIO 20 and 19). Everything works fine. I use a REPL terminal and control my ESP with mpremote.

However, I modified the firmware and I am not able to communicate with my esp32 using esptool, the error is as follows: 

`A fatal error occurred: Failed to connect to ESP32-S3: No serial data received.`

Adding the verbose gives me the following details: 

```
TRACE +0.000 Write 46 bytes: 
    c000082400000000 0007071220555555 | ...$........ UUU
    555555555555555555 5555555555555555 | UUUUUUUUUUUUUUUUUUUU
    5555555555555555 555555555555c0 | UUUUUUUUUUUU.
TRACE +0.105 No serial data received.
```

I don't have a REPL terminal open but I guess Micropython uses the USB interface  and I don't have access with esptool.

Also, on my board I don't have access to any other interface (UART, ...). I don't have a reset button either (can't access to GPIO 0)

Is there a way to disable Micropython from a REPL terminal to free the USB interface?

Thanks in advance!

**Comment by jimmo** [comment]
Maybe putting it into the bootloader manually will help: https://docs.espressif.com/projects/esptool/en/latest/esp32s3/advanced-topics/boot-mode-selection.html#boot-mode

**Reply by Nico-GYW** [reply]
Thank you for your answer. Unfortunately, I don't have access to the GPIO0 on this homemade board.`

A solution would be to enter DFU bootloader from software. I saw that Micropython has the function `machine.bootloader([value])` which resets the device and enter its bootloader. However, this function is not available on esp32. 

I also saw that this operation was done on an esp32-S2 with CircuitPython (https://github.com/adafruit/circuitpython/issues/3884).

Is there a way to force the bootloader state with a software command? 

On the esp32, Micropython also allows to erase the flash. A hard solution would be to use the function `esp.flash_erase(sector_no)` and delete the micropython binary (which in my case starts at address 0x10000) ? Is there any security that would make the device restart in boot mode?

Thanks
